tiledb/common/interval/test/unit_interval.h

```diff
@@ -364,7 +364,7 @@ Catch::Generators::GeneratorWrapper<typename ChooseGenerator<T>::value_type>
 choose(unsigned int k, std::initializer_list<T> list) {
   return Catch::Generators::GeneratorWrapper<
       typename ChooseGenerator<T>::value_type>(
-      std::make_unique<ChooseGenerator<T>>(k, list));
+      Catch::Detail::make_unique<ChooseGenerator<T>>(k, list));
```

Comment:
What the reason for this change?

Reply:
compiles vs not compiling :).
avoiding the following (maybe you have another form that will also avoid issue that you like better),
(think original 'hint' for that change was from here, https://github.com/catchorg/Catch2/blob/devel/examples/300-Gen-OwnGenerator.cpp)
```
  unit_interval_types.cc
D:\dev\tiledb\gh.sc-19871-upd-to-catch2-ver3.git\tiledb\common\interval\test\unit_interval.h(366,47): error C2440: '<function-style-cast>': cannot convert from 'std::unique_ptr<ChooseGenerator<_Ty>,std::default_delete<ChooseGenerator<_Ty>>>' to 'Catch::Generators::GeneratorWrapper<std::vector<T,std::allocator<T>>>' [D:\dev\tiledb\gh.sc-19871-upd-to-catch2-ver3.git\bld.vs22.D\tiledb\tiledb\common\interval\unit_interval.vcxproj]
          with
          [
              _Ty=T
          ] (compiling source file D:\dev\tiledb\gh.sc-19871-upd-to-catch2-ver3.git\tiledb\common\interval\test\unit_interval_types.cc)
D:\dev\tiledb\gh.sc-19871-upd-to-catch2-ver3.git\tiledb\common\interval\test\unit_interval.h(364,1): message : No constructor could take the source type, or constructor overload resolution was ambiguous (compiling source file D:\dev\tiledb\gh.sc-19871-upd-to-catch2-ver3.git\tiledb\common\interval\test\unit_interval_types.cc) [D:\dev\tiledb\gh.sc-19871-upd-to-catch2-ver3.git\bld.vs22.D\tiledb\tiledb\common\interval\unit_interval.vcxproj]
D:\dev\tiledb\gh.sc-19871-upd-to-catch2-ver3.git\tiledb\common\interval\test\unit_interval_types.cc(36): message : see reference to function template instantiation 'Catch::Generators::GeneratorWrapper<std::vector<T,std::allocator<T>>> choose<_Ty>(unsigned int,std::initializer_list<_Ty>)' being compiled [D:\dev\tiledb\gh.sc-19871-upd-to-catch2-ver3.git\bld.vs22.D\tiledb\tiledb\common\interval\unit_interval.vcxproj]
          with
          [
              _Ty=T
          ]
D:\dev\tiledb\gh.sc-19871-upd-to-catch2-ver3.git\tiledb\common\interval\test\unit_interval_types.cc(36): message : see reference to function template instantiation 'void CATCH2_INTERNAL_TEMPLATE_TEST_1<uint16_t>(void)' being compiled [D:\dev\tiledb\gh.sc-19871-upd-to-catch2-ver3.git\bld.vs22.D\tiledb\tiledb\common\interval\unit_interval.vcxproj]
D:\dev\tiledb\gh.sc-19871-upd-to-catch2-ver3.git\tiledb\common\interval\test\unit_interval_types.cc(36): message : while compiling class template member function 'void `anonymous-namespace'::ns_CATCH2_INTERNAL_TEMPLATE_TEST_0::CATCH2_INTERNAL_TEMPLATE_TEST_0<uint16_t,uint64_t,int16_t,double>::reg_tests(void)' [D:\dev\tiledb\gh.sc-19871-upd-to-catch2-ver3.git\bld.vs22.D\tiledb\tiledb\common\interval\unit_interval.vcxproj]
D:\dev\tiledb\gh.sc-19871-upd-to-catch2-ver3.git\tiledb\common\interval\test\unit_interval_types.cc(36): message : see reference to function template instantiation 'void `anonymous-namespace'::ns_CATCH2_INTERNAL_TEMPLATE_TEST_0::CATCH2_INTERNAL_TEMPLATE_TEST_0<uint16_t,uint64_t,int16_t,double>::reg_tests(void)' being compiled [D:\dev\tiledb\gh.sc-19871-upd-to-catch2-ver3.git\bld.vs22.D\tiledb\tiledb\common\interval\unit_interval.vcxproj]
D:\dev\tiledb\gh.sc-19871-upd-to-catch2-ver3.git\tiledb\common\interval\test\unit_interval_types.cc(36): message : see reference to class template instantiation '`anonymous-namespace'::ns_CATCH2_INTERNAL_TEMPLATE_TEST_0::CATCH2_INTERNAL_TEMPLATE_TEST_0<uint16_t,uint64_t,int16_t,double>' being compiled [D:\dev\tiledb\gh.sc-19871-upd-to-catch2-ver3.git\bld.vs22.D\tiledb\tiledb\common\interval\unit_interval.vcxproj]
D:\dev\tiledb\gh.sc-19871-upd-to-catch2-ver3.git\tiledb\common\interval\test\unit_interval.h(366,47): error C2440: '<function-style-cast>': cannot convert from 'std::unique_ptr<ChooseGenerator<_Ty>,std::default_delete<ChooseGenerator<_Ty>>>' to 'Catch::Generators::GeneratorWrapper<std::vector<size_t,std::allocator<size_t>>>' [D:\dev\tiledb\gh.sc-19871-upd-to-catch2-ver3.git\bld.vs22.D\tiledb\tiledb\common\interval\unit_interval.vcxproj]
          with
          [
              _Ty=size_t
          ] (compiling source file D:\dev\tiledb\gh.sc-19871-upd-to-catch2-ver3.git\tiledb\common\interval\test\unit_interval_types.cc)

```